content/blog/atomic-design-for-spas.md

```diff
@@ -0,0 +1,213 @@
+
+## Atomic Design
+
+If you have read the article introducing Atomic Design, you can probably guess that your `components` folder is going to look like that:
+
+```bash
+.
+├── atoms
+├── molecules
+├── organisms
+├── templates
+└── pages
+```
+
+Let's look at them in the same order.
+
+### Atoms
+
+Atoms are independent components that do not have any dependencies on other components and that exist in isolation: they
+probably don't do anything on their own, are mostly composed of `props` and any action they do is coming from `props`.
+In terms of Redux/MobX, all of those components should be dumb/pure/presentational components.
+
+Looking at one private project atoms, I have:
+
+- `button`
+- `input`
+- `link`
+- `tag`
+- `toast`
+- `portal`
+- `icon`
+
+There are obviously many more but this should be enough to give you an idea: they are the building blocks
+of your site.
+
+Try to have as many of your components in that folder, you will have a much easier time composing interfaces later on.
+
+### Molecules
+
+By themselves, atoms are not very useful and need to be combined to make something useable: `input` + `button` giving you a `form` for example.
+That is what molecules are: a combination of 2 or more `atoms` or, in rarer cases, a smart wrapper around an atom.
+Therefore molecules should be dumb components when possible but it isn't required.
+
+Here are some of the molecules from the same project:
+
+- `form` (`input` + `label`): using MobX for state but dumb otherwise
+- `dialog` (`overlay` + `button`): dumb
+- `toast-container` (`toast`): smart component that will display the currents `<Toast>` components from the store
+
+Whether `toast-container` belongs in the molecules folder is debatable
+but this is where it makes the most sense in my opinion.
+
+If you need to use a molecule in another molecule, it probably means it should go in the organisms folder as we will now see.
+
+### Organisms
+
+As with molecules, there are two types of organisms I distinguish here:
+
+- components using one or more molecules
+- business-specific usage of molecule
+
+Another good definition comes from the original article:
+
+>  Organisms are groups of molecules joined together to form a relatively complex, distinct section of an interface.
+
+As before, here are some examples in my project:
+
+- `sidebar`: uses only one molecule but is a distinct section of the interface
+- `login-form`: the `form` molecule set-up with the right fields and the GraphQL login query
+- `signup-form`: the `form` molecule set-up with the right fields and the GraphQL signup query
+
+As you can see, each form in the app gets its own component making testing/storybook very easy.
+Most of those forms components do not have style of their own as they just rely on the `form` molecule for that.
+
+### Templates
+
+My usage of templates is probably the biggest deviation from the original Atomic Design. If you look at most SaaS sites, you will notice
```

Comment:
Is it different? from my understanding, skimming the atomic design article, a template is a collection of organisms, with a layout. It seems to be exactly what you're doing